distribution/pull_v2.go

```diff
@@ -367,13 +369,34 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 			"remote": ref,
 		}))
 
-	manifest, err := p.manifestStore.Get(ctx, specs.Descriptor{
+	desc := specs.Descriptor{
 		MediaType: mt,
 		Digest:    dgst,
 		Size:      size,
-	})
+	}
+	manifest, err := p.manifestStore.Get(ctx, desc)
 	if err != nil {
-		return false, err
+		if isTagged && isNotFound(errors.Cause(err)) {
+			logrus.WithField("ref", ref).WithError(err).Debug("Falling back to pull manifest by tag")
+
+			progress.Message(p.config.ProgressOutput, tagOrDigest, "Failed to pull manifest by the resolved digest")
+			progress.Message(p.config.ProgressOutput, tagOrDigest, "This registry does not appear to conform to the distribution specificiation")
```

Comment:
should we start the message with something like a big "DEPRECATED:" ?

Reply:
Added.